compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -610,6 +610,16 @@ impl<'a> Parser<'a> {
         // FIXME: translation requires list formatting (for `expect`)
         let mut err = self.struct_span_err(self.token.span, msg_exp);
 
+        // Look for usages of '=>' where '>=' was probably intended
+        if self.token == token::FatArrow {
```

Comment:
I think that changing this to
```suggestion
        if self.token == token::FatArrow
            && expected.iter().any(|tok| matches!(tok, TokenType::Token(TokenKind::Le)))
```
will help quite a bit in reducing the false positives. Can you make that change and see what the results in the test suite are?

Reply:
Looking at all of these is it's making me think it'd be nice to also detect [`match foo { pat >= {} }`](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=f886f820f0968190e6145fb084782115) (but that likely is less confusing) 😅

Reply:
> Can you make that change and see what the results in the test suite are?

Unfortunately that misses valid cases like `let _ = a => b`. I also tried `self.token == token::FatArrow && expected.contains(&TokenType::Operator)`, but that was too broad and flagged invalid cases like `asm!("{}", in(reg) foo => bar);` (from [this set of tests](https://github.com/rust-lang/rust/blob/d4c86cfc4984293d642e5b4f7eece60a6e83e264/tests/ui/asm/parse-error.rs#L29)).